From e4b4c26451c296837cae771d7732182ec36255ae Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Mon, 24 Feb 2020 17:56:27 +0800
Subject: [PATCH] soc: rockchip: opp_select: Fix compilation warnings when
 !CONFIG_ROCKCHIP_OPP

include/soc/rockchip/rockchip_opp_select.h: In function 'rockchip_set_opp_prop_name':
include/soc/rockchip/rockchip_opp_select.h:83:9: warning: returning 'int' from a function with return type 'struct opp_table *' makes pointer from integer without a cast [-Wint-conversion]

Change-Id: Id8c0fc331bcece33692ddde2084a7e8d3142bd94
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 include/soc/rockchip/rockchip_opp_select.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/soc/rockchip/rockchip_opp_select.h b/include/soc/rockchip/rockchip_opp_select.h
index 97f8d3e4a9ca..5c4abed9ecec 100644
--- a/include/soc/rockchip/rockchip_opp_select.h
+++ b/include/soc/rockchip/rockchip_opp_select.h
@@ -80,7 +80,7 @@ static inline struct opp_table *rockchip_set_opp_prop_name(struct device *dev,
 							   int process,
 							   int volt_sel)
 {
-	return -ENOTSUPP;
+	return ERR_PTR(-ENOTSUPP);
 }
 
 static inline int rockchip_adjust_power_scale(struct device *dev, int scale)
-- 
2.35.3

